usage/test/org/apache/cloudstack/quota/QuotaAlertManagerImplTest.java

```diff
@@ -0,0 +1,239 @@
+
+import com.cloud.domain.DomainVO;
+import com.cloud.domain.dao.DomainDao;
+import com.cloud.user.Account;
+import com.cloud.user.AccountVO;
+import com.cloud.user.UserVO;
+import com.cloud.user.dao.AccountDao;
+import com.cloud.user.dao.UserDao;
+import com.cloud.utils.db.TransactionLegacy;
+import junit.framework.TestCase;
+import org.apache.cloudstack.framework.config.dao.ConfigurationDao;
+import org.apache.cloudstack.quota.constant.QuotaConfig;
+import org.apache.cloudstack.quota.dao.QuotaAccountDao;
+import org.apache.cloudstack.quota.dao.QuotaEmailTemplatesDao;
+import org.apache.cloudstack.quota.dao.QuotaUsageDao;
+import org.apache.cloudstack.quota.vo.QuotaAccountVO;
+import org.apache.cloudstack.quota.vo.QuotaEmailTemplatesVO;
+import org.joda.time.DateTime;
+import org.junit.Before;
+import org.junit.Test;
+import org.junit.runner.RunWith;
+import org.mockito.Mock;
+import org.mockito.Mockito;
+import org.mockito.Spy;
+import org.mockito.runners.MockitoJUnitRunner;
+
+import javax.mail.MessagingException;
+import javax.naming.ConfigurationException;
+import java.io.UnsupportedEncodingException;
+import java.lang.reflect.Field;
+import java.math.BigDecimal;
+import java.util.ArrayList;
+import java.util.Calendar;
+import java.util.Date;
+import java.util.List;
+
+@RunWith(MockitoJUnitRunner.class)
+public class QuotaAlertManagerImplTest extends TestCase {
+
+    @Mock
+    AccountDao accountDao;
+    @Mock
+    QuotaAccountDao quotaAcc;
+    @Mock
+    UserDao userDao;
+    @Mock
+    DomainDao domainDao;
+    @Mock
+    QuotaEmailTemplatesDao quotaEmailTemplateDao;
+    @Mock
+    ConfigurationDao configDao;
+    @Mock
+    QuotaUsageDao quotaUsage;
+    @Mock
+    QuotaAlertManagerImpl.EmailQuotaAlert emailQuotaAlert;
+
+    @Spy
+    QuotaAlertManagerImpl quotaAlertManager = new QuotaAlertManagerImpl();
+
+    private void injectMockToField(Object mock, String fieldName) throws NoSuchFieldException, IllegalAccessException {
+        Field f = QuotaAlertManagerImpl.class.getDeclaredField(fieldName);
+        f.setAccessible(true);
+        f.set(quotaAlertManager, mock);
+    }
+
+    @Before
+    public void setup() throws IllegalAccessException, NoSuchFieldException, ConfigurationException {
+        // Dummy transaction stack setup
+        TransactionLegacy.open("QuotaAlertManagerImplTest");
+
+        injectMockToField(accountDao, "_accountDao");
+        injectMockToField(quotaAcc, "_quotaAcc");
+        injectMockToField(userDao, "_userDao");
+        injectMockToField(domainDao, "_domainDao");
+        injectMockToField(quotaEmailTemplateDao, "_quotaEmailTemplateDao");
+        injectMockToField(configDao, "_configDao");
+        injectMockToField(quotaUsage, "_quotaUsage");
+        injectMockToField(emailQuotaAlert, "_emailQuotaAlert");
+    }
+
+    @Test
+    public void testStartStop() {
+        try {
+            quotaAlertManager.start(); // expected to fail as pid is not available
```

Comment:
So it should fail if the ``start`` call did not throw an exception.  Therefore, immediately following that call to start, you should call ``fail`` because it should never reach that point in the code if an exception is thrown.